server/routes/transferRouter.js

```diff
@@ -151,10 +167,32 @@ transferRouter.get("/",
   helper.apiKeyHandler,
   helper.verifyJWTHandler,
   helper.handlerWrapper(async (req, res) => {
+    Joi.assert(
+      req.query,
+      Joi.object({
+        state: Joi.string()
+          .valid(...Object.values(Transfer.STATE)),
+        wallet: Joi.alternatives().try(
+          Joi.string().alphanum().min(4).max(32),
+          Joi.number().min(4).max(32)
+        ),
+      })
+    );
     const {state, wallet} = req.query;
     const walletService = new WalletService();
     const walletLogin = await walletService.getById(res.locals.wallet_id);
-    const result = await walletLogin.getTransfers(state, wallet);
+
+    let walletObject;
+    if(wallet){
+      if (isNaN(wallet)) {
+        walletObject = await walletService.getByName(wallet);
+      } else {
+        walletObject = await walletService.getById(wallet);
+      }
+    }
```

Comment:
If this logic will be used frequently, should we store it as a function in utils that will return the appropriate walletObject?